### DISCUSSION TITLE
[Documentation]: How to add custom MDX components

### DISCUSSION THREAD

**Discussion by btmnk** [storybookjs/storybook#25355] [Help]
### Describe the problem

I was expecting to find some guidance on how to add custom mdx components at https://storybook.js.org/docs/react/writing-docs/mdx but I couldn't find it. In some issues I read that there should be something like the following possible:

```ts
const preview: Preview = {
  parameters: {
    docs: {
      components: {
        pre: MdxPre,
        h1: Test,
      },
    },
    ...
  },
};
```

Where MdxPre and Test are just normal React Components.
But they are just completely ignored. (Unfortunately the Preview interface is also quite useless as it doesn't give any type safety at all, I assume that's because most of the configuration actually comes from addons and might not be present)

I'm not sure if this is still possible with v7 or not as the documentation lacks any information about it.
At the very least the documentation should explain something whether its possible or not.

Even better would be a clear setup for mdx that is not hidden behind untyped storybook configuration.
(I also wasn't sure how to add rehype plugins since the docs only show examples of remark plugins)

### Additional context

I'm currently on v7.6.0-alpha.2 so it's possible that is why it's not working but nontheless documentation would be helpful to be sure where to add components.

**Comment by TinsFox** [comment]
I also need this functionality, although we can import in each mdx file, this may not be the best method.
